activestorage/lib/active_storage/attached/model.rb

```diff
@@ -144,7 +144,7 @@ def #{name}=(attachables)
                 if attachables.none?
                   ActiveStorage::Attached::Changes::DeleteMany.new("#{name}", self)
                 else
-                  ActiveStorage::Attached::Changes::CreateMany.new("#{name}", self, attachables)
+                  ActiveStorage::Attached::Changes::CreateMany.new("#{name}", self, attachables, pending_uploads: attachment_changes["#{name}"].try(:pending_uploads))
```

Comment:
Hey @santib I think we can extract a variable here outside of the `if` to remove the duplication:

```ruby
pending_uploads = attachment_changes["#{name}"].try(:pending_uploads)
```

Then you can reuse it in both invocations: `... pending_uploads: pending_uploads`.

With that in place I'll get this merged I promise 😊. Thanks a lot for your work here!

Reply:
of course! no worries 😄

Reply:
Great thanks. Cc @Jeremy to merge